src/inmanta/module.py

```diff
@@ -426,7 +434,26 @@ def get_module(
         """
         module_name: str = self._get_module_name(module_spec)
         installed: Optional[TModule] = self.get_installed_module(project, module_name)
-        if installed is None and install:
+
+        def _should_install_module() -> bool:
+            """
+            Return True iff the given module should get installed
+            """
+            if not install:
+                # No install was requested
+                return False
+            if installed is None:
+                # Package is not installed
+                return True
+            if isinstance(installed, ModuleV2):
+                pip_requirements = [r.get_pip_requirement() for r in module_spec]
+                if not project.virtualenv.are_installed(pip_requirements):
+                    # Package could define an extra that is not installed yet
+                    return True
```

Comment:
This has the side effect of also reinstalling the module if its constraints are not met. While I at first glance think this is a good thing, we need to carefully consider why we didn't do so in the first place (we only checked for `installed is None`) and whether it might have adverse effects. I need to think about this some more.

Reply:
I think this should be fine. And we've got a whole release cycle to figure it out if it turns out otherwise.

Reply:
The docstring for module_spec parameter should be updated `Ignored if module
            is already installed` is no longer true.